### ISSUE TITLE
Issue finding Unity on Windows

### ISSUE THREAD

**Issue by SbastianGarzon** [ropensci/unifir#13]
I'm having issues finding Unity on Windows:
```
library(unifir)
unity_path <- file.path("C:","Program Files","Unity","Editor")
file.exists(unity_path)

[1] TRUE

options(unifir_unity_path=unity_path)
Sys.getenv("unifir_unity_path")

[1] "c:/Program files/Unity/Editor"

unity_version()
Error in system(paste(unity, "-version"), intern = TRUE) : 
  'c:/Program' not found

```
Here is my session info:

```
R version 4.2.2 (2022-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=Dutch_Netherlands.utf8  LC_CTYPE=Dutch_Netherlands.utf8    LC_MONETARY=Dutch_Netherlands.utf8
[4] LC_NUMERIC=C                       LC_TIME=Dutch_Netherlands.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] unifir_0.2.2

loaded via a namespace (and not attached):
[1] compiler_4.2.2 R6_2.5.1       tools_4.2.2 
```

**Comment by mikemahoney218** [comment]
Thanks for the report!

Sorry, this package is primarily tested on Linux and Mac, which means I forget about things like Windows file paths typically having spaces in them. 

Any chance you can install my attempted solution and tell me if this fixed things?
```r
remotes::install_github("mikemahoney218/unifir", "fixes")
```

**Comment by SbastianGarzon** [comment]
Thanks !

Now I'm getting this:
```
unity_version()
Error in system(paste(shQuote(unity), "-version", intern =TRUE):
'"C:Program Files/Unity/Editor"' not found
```
(Seems better but still not finding Unity)

I changed the `unifir_unity_path` variable to `file.path("C:","Program Files","Unity","Editor","Unity.exe")` and it seems to be working (kind of).

```
start_unity <-Sys.time()
unity_version()
character(0)
end_unity <- Sys.time()
Time difference of 12.29863 secs
```
After using the unity_version() command, the Unity Hub application opens in my computer (good sign ?)

However this happens as I tried your example:

```
script <- make_script(project="myproj")
script <-add_default_player(script,x_position=10)
script <- add_light(script)
action(script)
Error in action(script): 1
```

I might be missing something obvious about what to do with the `unifir_unity_path` option. I'm also having a similar problem with Mac to connect to Unity, so that's why I tried windows.

**Comment by SbastianGarzon** [comment]
I'm going to close this issue. The "finding Unity on Windows" problem is actually fixed.  The new error might not be related with finding Unity. 

For example this works:
```
script$script_name
[1] "HappenGoneSteadPath"
```

**Comment by mikemahoney218** [comment]
Yeah, I apologize -- unifir is not _really_ meant to be used directly, but rather to prop up things like [terrainr](https://github.com/ropensci/terrainr) and an unreleased forest rendering project I'm working on. That means that the docs aren't exactly _great_ for interactive use; they sorta assume users will use one of those not-released projects instead.

That said: In a new session, does `find_unity()` return the correct path to your Unity installation? It looks like `action()` ran successfully on your computer (though _Unity itself failed_, hence the error message -- that error code is literally the only output Unity returns when run via CLI, which is a pain), which means instead the issue is with the example.

**Comment by SbastianGarzon** [comment]
No problem. I'll patiently wait for the new release 🌲🌲🌲 ! I'll try terrainr to see if I manage to create an Unity project.

Here is the output in a new session:

```
find_unity()
[1] "C:/Program Files/Unity/Editor/Unity.exe"
```
When I run `action()` some folders/files are created before failing 🤷‍♂️:
`Assets`, `myproj`,`Library`, `ProjectSettings` and `Temp`.

**Comment by mikemahoney218** [comment]
So far you've helped me fix three bugs, thank you :smile: 

The proximate issue here is that `add_default_player()` didn't import a C# library it needed, which means it failed immediately. This is now fixed on the `fixes` branch, and will be headed to CRAN in the immediate future (because I need to cut a release soon to fix #12 anyway).

There's one other bug in the example: you need to call `save_scene()`, and probably _want_ to call `set_active_scene()` afterwards, in order for your changes to appear once you load Unity. Otherwise the unifir script will make all your edits, and then not save any of them. I'm going to open a new issue for a documentation refresh; hopefully I'm able to make the new user experience for unifir itself better in Q1'2023, while also providing more easy-to-use front-ends like terrainr.
